sn_client/src/audit/spend_dag.rs

```diff
@@ -457,7 +461,7 @@ impl SpendDag {
                     });
                     let actual_ancestor: Vec<_> = multiple_ancestors
                         .iter()
-                        .filter(|(s, _)| s.spend.spent_tx.hash() == spend.spend.parent_tx.hash())
+                        .filter(|(s, _)| s.address() == spend.address())
                         .map(|(s, _)| s.clone())
                         .collect();
```

Comment:
This check should check that the `s` is indeed the parent of `spend` by checking that:
```rust
s.spend.descendant.contains(spend.address())
&&
spend.spend.parent.contains(s.address())
```